radicle-node/src/service.rs

```diff
@@ -860,14 +871,31 @@ where
                         match message.is_fresh(&self.storage) {
                             Ok(is_fresh) => {
                                 if is_fresh {
-                                    // TODO: Only fetch if the refs announced are for peers we're tracking.
-                                    self.fetch(message.rid, announcer);
+                                    let repo_tracking = self
+                                        .tracking
+                                        .repo_entry(&message.rid)
+                                        .expect("handle_announcement: error accessing repo tracking configuration");
```

Comment:
We're already accessing the tracking policy in the parent `if` statement, so maybe bind this result outside and it can be used here.

Reply:
yeah, the previous access is via `is_repo_tracked()` that hides the repo_entry internally. Let me see what we can do.

Reply:
I've refactored this part of code to bind the store result first, and then use it in both places.